sci-wcalc/ic_microstrip_calc.man

.\" Copyright (c), 2005 Dan McMahill
.\" Do not edit this directly.  Edit the XML source file instead
.\"

.TH ic_microstrip_calc "" "" "Wcalc" "Wcalc Commands"
.B ic_microstrip_calc
- Integrated circuit microstrip transmission line analysis

.SH SYNOPSIS

[z0, keff, elen, loss, L, R, C, G] = 
ic_microstrip_calc(w, l, tox, h, er, sigmas, tmet, rho, rough, f)


.SH INPUT PARAMETERS

.TP 10
w
: width of ic_microstrip line (meters)
.TP 10
l
: lentgh of ic_microstrip line (meters)
.TP 10
tox
: oxide thickness (meters)
.TP 10
h
: substrate thickness (meters)
.TP 10
er
: substrate relative permitivity
.TP 10
sigmas
: substrate conductivity (1/ohm-m)
.TP 10
tmet
: metal thickness (meters)
.TP 10
rho
: resistivity of wire (relative to copper)
.TP 10
rough
: conductor surface roughness (meters-rms)
.TP 10
f
: synthesis frequency (Hz)

.SH OUTPUT VALUES

.TP 10
z0
: calculated characteristic impedance (Ohms)
.TP 10
keff
: calculated effective relative permitivitty
.TP 10
elen
: calculated electrical length of the line (degrees)
.TP 10
loss
: calculated loss of the line (dB)
.TP 10
L
: calculated inductance per length (Henries/meter)
.TP 10
R
: calculated resistance per length (Ohms/meter)
.TP 10
C
: calculated capacitance per length (Farads/meter)
.TP 10
G
: calculated conductance per length (Siemens/meter)
.SH DESCRIPTION

Function to analyze the electrical characteristics of a single 
integrated circuit microstrip transmission line. The physical
dimensions of the microstrip line are given as inputs and the
characteristic impedance, effective dielectric constant, loss and
open circuit end correction are calculated.


.NF



                 |<--W-->|
                  _______    
                 | metal | <- tmet,rho,rough
    ----------------------------------------------
   {**************  Oxide (tox,eox) **************
    ----------------------------------------------
   (  substrate                          /|\     (
    )   es,sigmas                     H   |       )
   (                                     \|/     (
    ----------------------------------------------
    XXXXXXXXXXXXXXXXXX ground XXXXXXXXXXXXXXXXXXXX



.FI

.SH EXAMPLE
.nf

.TP
 // width
w=150e-6;
.TP
 // substrate height
h=250e-6;
.TP
 // length
l=1000e-6;
.TP
 // substrate permitivitty
es=11.8;
.TP
 // substrate conductivity
sigmas=0.5;
.TP
 // oxide thickness
tox=1.5e-6;
.TP
 // oxide permitivitty
eox=4;
.TP
 // metal thickenss
tmet=0.8e-6;
.TP
 // metal resitivity
rho=3.0e-8;
.TP
 // metal surface roughness
rough=0.0;
.TP
 // frequency
f=2.4e9;
[z0,keff,elen,loss,L,R,C,G] = ...
  ic_microstrip_calc(w,l,tox,eox,h,es,sigmas,tmet,rho,rough,f);
disp(sprintf('z0 = %8.4g + j %8.4g Ohms', real(z0), imag(z0) ));
disp(sprintf('keff = %8.4g ',keff));
disp(sprintf('elen = %8.4g degrees',elen));
disp(sprintf('loss = %8.4g dB',loss));
disp(sprintf('L  = %8.4g nH/mm', L*1e6));
disp(sprintf('R  = %8.4g Ohm/mm', R*1e-3));
disp(sprintf('C  = %8.4g fF/mm', C*1e12));
disp(sprintf('G  = %8.4g uMho/mm', G*1e3));
.fi
.SH SEE ALSO
stdio-wcalc(1),
air_coil_calc(), air_coil_syn(), coax_calc(), coax_syn(), coplanar_calc(), coplanar_syn(), coupled_microstrip_calc(), coupled_microstrip_syn(), coupled_stripline_calc(), coupled_stripline_syn(), ic_microstrip_calc(), ic_microstrip_syn(), microstrip_calc(), microstrip_syn(), stripline_calc(), stripline_syn()
.SH AUTHOR

Dan McMahill

.SH BUGS

None known


NEWS
$Id: NEWS,v 1.10 2009/02/28 03:26:18 dan Exp $

Wcalc -- history of user-visible changes. 

Copyright (c) 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005
2006, 2007, 2008, 2009 Dan McMahill

Please report bugs on the sourceforge project page.

===============
Version 1.1
===============

Added series/parallel RC equivalent circuit calculator.

Added series/parallel RL equivalent circuit calculator.

Added self and mutual inductance of two rectangular bars.

Added coplanar waveguide model.

Added coupled stripline model.

Corrected the Q calculations for low frequency (not skindepth) region
in air core inductors.  Also corrected Q calculation for conductors
other than copper.

Corrected the calculation of incremental conductance for the
microstrip model.

Corrected a bug in the calculation of conductor losses in the
microstrip model.

Added/enabled code for calculating conductor losses in the coupled
microstrip model.

When compiling with gtk-2.10 or newer, use gtkPrint for printing.
This gives a more professional/standard print dialog and also enables
printing under windows.

Added a preliminary Dutch translation.

Converted the figures to all use solid grayscale fills instead of
pattern fills as the former prints better.

Several updates to the build system to moderize it a bit.  Uses
AM_CONDITIONALS more appropriately, uses AC_PROG_LIBTOOL instead of
AM_PROG_LIBTOOL, etc.

Update the win32 build script to generate an installer that works
correctly under vista.

Add .wc file associations under vista.

Update the octave build infrastructure to work with octave-3.0 and
newer where mex support is built into mkoctfile instead of a
standalone mex shell script.

Switch to GPL licensing.

Create several utility functions to make it faster to build new gtk
forms.  This is only visible to developers.

Improved autogen.sh to check versions of several of the auto* tools to
help make it easier to diagnose any issues which may come up.

Repair compilation with the SunPRO compilers.


===============
Version 1.0
===============
Replaced the alert dialog box in the gtk version with a log
window.  This prevents a problem when many alerts are generated
and the user is presented with many many alert windows.

Added win32 support along with a build script for a windows installer.

Fixed a bug which gave lots of error/warning messages in the cgi
frontend for coupled microstrip.

===============
Version 0.9
===============
First public release.

===============
Version 0.1 
===============
Initial tagged version.  This does good analysis and synthesis
of microstrip.  Considered stable for this purpose.  However,
open/save/print and units other than mils are not supported yet.



octave-wcalc/ic_microstrip_calc.m
%IC_MICROSTRIP_CALC - Integrated circuit microstrip transmission line analysis 
%  
%  SYNOPSIS
%  
%  [z0, keff, elen, loss, L, R, C, G] = ic_microstrip_calc(w, l, tox, h, er,
%  sigmas, tmet, rho, rough, f)
%  
%  PARAMETERS
%  
%  INPUT PARAMETERS
%  
%    * w : width of ic_microstrip line (meters)
%    * l : lentgh of ic_microstrip line (meters)
%    * tox : oxide thickness (meters)
%    * h : substrate thickness (meters)
%    * er : substrate relative permitivity
%    * sigmas : substrate conductivity (1/ohm-m)
%    * tmet : metal thickness (meters)
%    * rho : resistivity of wire (relative to copper)
%    * rough : conductor surface roughness (meters-rms)
%    * f : synthesis frequency (Hz)
%  
%  OUTPUT VALUES
%  
%    * z0 : calculated characteristic impedance (Ohms)
%    * keff : calculated effective relative permitivitty
%    * elen : calculated electrical length of the line (degrees)
%    * loss : calculated loss of the line (dB)
%    * L : calculated inductance per length (Henries/meter)
%    * R : calculated resistance per length (Ohms/meter)
%    * C : calculated capacitance per length (Farads/meter)
%    * G : calculated conductance per length (Siemens/meter)
%  
%  DESCRIPTION
%  
%  Function to analyze the electrical characteristics of a single integrated
%  circuit microstrip transmission line. The physical dimensions of the microstrip
%  line are given as inputs and the characteristic impedance, effective dielectric
%  constant, loss and open circuit end correction are calculated.
%  
%  
%                   |<--W-->|
%                    _______
%                   | metal | <- tmet,rho,rough
%      ----------------------------------------------
%     {**************  Oxide (tox,eox) **************
%      ----------------------------------------------
%     (  substrate                          /|\     (
%      )   es,sigmas                     H   |       )
%     (                                     \|/     (
%      ----------------------------------------------
%      XXXXXXXXXXXXXXXXXX ground XXXXXXXXXXXXXXXXXXXX
%  
%  
%  EXAMPLE
%  
%   % width
%  w=150e-6;
%   % substrate height
%  h=250e-6;
%   % length
%  l=1000e-6;
%   % substrate permitivitty
%  es=11.8;
%   % substrate conductivity
%  sigmas=0.5;
%   % oxide thickness
%  tox=1.5e-6;
%   % oxide permitivitty
%  eox=4;
%   % metal thickenss
%  tmet=0.8e-6;
%   % metal resitivity
%  rho=3.0e-8;
%   % metal surface roughness
%  rough=0.0;
%   % frequency
%  f=2.4e9;
%  [z0,keff,elen,loss,L,R,C,G] = ...
%    ic_microstrip_calc(w,l,tox,eox,h,es,sigmas,tmet,rho,rough,f);
%  disp(sprintf('z0 = %8.4g + j %8.4g Ohms', real(z0), imag(z0) ));
%  disp(sprintf('keff = %8.4g ',keff));
%  disp(sprintf('elen = %8.4g degrees',elen));
%  disp(sprintf('loss = %8.4g dB',loss));
%  disp(sprintf('L  = %8.4g nH/mm', L*1e6));
%  disp(sprintf('R  = %8.4g Ohm/mm', R*1e-3));
%  disp(sprintf('C  = %8.4g fF/mm', C*1e12));
%  disp(sprintf('G  = %8.4g uMho/mm', G*1e3));
%  
%  SEE ALSO
%  
%  air_coil_calc
%  air_coil_syn
%  coax_calc
%  bars_calc
%  coax_syn
%  coplanar_calc
%  coplanar_syn
%  coupled_microstrip_calc
%  coupled_microstrip_syn
%  coupled_stripline_calc
%  coupled_stripline_syn
%  ic_microstrip_calc
%  ic_microstrip_syn
%  microstrip_calc
%  microstrip_syn
%  parallel_rc_p2s
%  parallel_rc_s2p
%  parallel_rl_p2s
%  parallel_rl_s2p
%  stripline_calc
%  stripline_syn
%  
%  AUTHOR
%  
%  Dan McMahill
%  
%  BUGS
%  
%  None known
%  
%  Please report any bugs on the Wcalc Sourceforge Project Page
%  


% Copyright (c) 2001-2009 Dan McMahill
% All rights reserved.
%
% This program is free software; you can redistribute it and/or modify
% it under the terms of the GNU General Public License as published by
% the Free Software Foundation; version 2 of the License.
% 
% This program is distributed in the hope that it will be useful,
% but WITHOUT ANY WARRANTY; without even the implied warranty of
% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
% GNU General Public License for more details.
% 
% You should have received a copy of the GNU General Public License
% along with this program; if not, write to the Free Software
% Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
%
